### PR TITLE
Statsd

### PR DESCRIPTION
**Pull request by bmarini**
This adds a new adapter for statsd. I altered the adapter interface to be more lower level, allowing me to move the metrics collection up to the `Stats` struct.

### CODE REVIEW

File: adapter.go [hunk 1]
```diff
@@ -44,130 +38,79 @@ func NewLogAdapter(tmpl string) (*LogAdapter, error) {
...
+		Value:     value,
+	}
+	fmt.Println(renderTemplate(a.template, data))
 }
 
-func (s stat) Hostname() string {
-	return hostname
-}
+// StatsdTemplate defines the template used to render the statsd metric name.
+var StatsdTemplate = `{{.Name}}.source__{{.Container.Name}}__`
```
Comment (ejholmes) [inline adapter.go]: We can potentially suffix the hostname here as well too?

```
.source__{{.Container.Name}}.{{.Hostname}}__
```

File: cmd/stats/main.go [hunk 2]
```diff
@@ -10,6 +11,16 @@ import (
 
 var flags = []cli.Flag{
 	cli.StringFlag{
+		Name:   "adapter",
+		Value:  "log",
+		EnvVar: "STAT_ADAPTER",
+	},
+	cli.StringFlag{
+		Name:   "statsd.address",
```
Comment (ejholmes) [inline cmd/stats/main.go]: What do you think about just pulling this from the `adapter` flag?

```
-adapter log://
```

```
-adapter statsd://localhost:8125
```

**Comment by ejholmes** [comment]
Nice. lgtm :+1:

File: cmd/stats/main.go (hunk 2 above)
Comment (bmarini) [inline cmd/stats/main.go]: that's cool

**Pull request merged**